src/bounds.rs

```diff
@@ -0,0 +1,224 @@
+                end,
+            },
+
+            // --start or --end is a date
+            (Some(Bound::Date(start)), Some(Bound::Date(end))) => {
+                check_in_future("start", &start)?;
+                check_in_future("end", &end)?;
+                Bounds::Dates { start, end }
+            }
+            (Some(Bound::Date(start)), None) => {
+                check_in_future("start", &start)?;
+                Bounds::Dates {
+                    start,
+                    end: find_latest_nightly()?,
+                }
+            }
+            (None, Some(Bound::Date(end))) => {
+                check_in_future("end", &end)?;
+                if args.by_commit {
+                    bail!("--by-commit with an end date requires --start to be specified");
+                }
+                Bounds::SearchNightlyBackwards { end }
+            }
+
+            // Mixed not supported.
+            (Some(Bound::Commit(_)), Some(Bound::Date(_)))
+            | (Some(Bound::Date(_)), Some(Bound::Commit(_))) => bail!(
+                "cannot take different types of bounds for start/end, \
+                 got start: {:?} and end {:?}",
+                args.start,
+                args.end
+            ),
+        };
+        if let Bounds::Dates { start, end } = &bounds {
+            if end < start {
+                bail!("end should be after start, got start: {start} and end {end}");
+            }
+            if args.by_commit {
+                eprintln!("finding commit range that corresponds to dates specified");
+                let bounds = Bounds::Commits {
+                    start: date_to_sha(&start)?,
+                    end: date_to_sha(&end)?,
+                };
+                return Ok(bounds);
+            }
+        }
+        Ok(bounds)
+    }
+}
+
+/// Translates a tag-like bound (such as `1.62.0`) to a `Bound::Date` so that
+/// bisecting works for versions older than 167 days.
+fn translate_tags(args: &Opts) -> anyhow::Result<(Option<Bound>, Option<Bound>)> {
+    let is_tag = |bound: &Option<Bound>| -> bool {
+        match bound {
+            Some(Bound::Commit(commit)) => commit.contains('.'),
+            None | Some(Bound::Date(_)) => false,
+        }
+    };
+    let is_datelike = |bound: &Option<Bound>| -> bool {
+        matches!(bound, None | Some(Bound::Date(_))) || is_tag(bound)
+    };
+    if !(is_datelike(&args.start) && is_datelike(&args.end)) {
+        // If the user specified an actual commit for one bound, then don't
+        // even try to convert the other bound to a date.
+        return Ok((args.start.clone(), args.end.clone()));
+    }
+    let fixup = |which: &str, bound: &Option<Bound>| -> anyhow::Result<Option<Bound>> {
+        if is_tag(bound) {
+            if let Some(Bound::Commit(tag)) = bound {
+                let date = args
+                    .access
+                    .repo()
+                    .bound_to_date(Bound::Commit(tag.clone()))?;
```

Comment:
This feels wrong to me. If we're looking up (say) 1.68.0, we presumably need something here to say that we want the *stable* release as of that date or otherwise go from the stable release to the nightly date range it "points" at?

Otherwise we're missing ~6 weeks here, right?

Reply:
The `RustRepositoryAccessor::commit` function returns the **merge base** with the master branch. So it essentially looks for the point where the given tag branched off the master branch. It does not return the commit associated with that tag.

That means it does not include any fixes merged to the `beta` branch, but should provide a close approximation of what the equivalent nightly was.

(This particular code is unchanged from the original, and has been working sufficiently well for me.)

Reply:
Ah, I missed that aspect. Sounds good to me.